test/tmetrics.m

```diff
@@ -0,0 +1,349 @@
+            verifyEqual(testCase, ct.Name, countername);
+
+            % create testing value and dictionary
+            dict = dictionary("k1","v1","k2",5);
+            vals = [1,2,3];
+            dict_keys = keys(dict);
+            dict_vals = values(dict);
+
+            % add value and attributes
+            ct.add(vals(1),dict);
+            ct.add(vals(2),dict);
+            ct.add(vals(3),dict_keys(1),dict_vals(1),dict_keys(2),dict_vals(2));
+
+            % wait for default collector response time (60s)
+            pause(70);
+
+            % fetch result
+            results = readJsonResults(testCase);
+            results = results{1};
+
+            % verify meter and counter names
+            verifyEqual(testCase, string(results.resourceMetrics.scopeMetrics.metrics.name), countername);
+            verifyEqual(testCase, string(results.resourceMetrics.scopeMetrics.scope.name), metername);
+
+            % fetch datapoint
+            dp = results.resourceMetrics.scopeMetrics.metrics.sum.dataPoints;
+
+            % verify counter value
+            verifyEqual(testCase, dp.asDouble, sum(vals));
+
+            % verify counter attributes
+            verifyEqual(testCase, string(dp.attributes(1).key), dict_keys(1));
+            verifyEqual(testCase, string(dp.attributes(1).value.stringValue), dict_vals(1));
+            verifyEqual(testCase, string(dp.attributes(2).key), dict_keys(2));
+            verifyEqual(testCase, string(dp.attributes(2).value.stringValue), dict_vals(2));
+
+        end
+
+
+        function testCounterAddNegative(testCase)
+            % test if counter value remain 0 when added negative value
+            
+            metername = "foo";
+            countername = "bar";
+
+            p = opentelemetry.sdk.metrics.MeterProvider();
+            mt = p.getMeter(metername);
+            ct = mt.createCounter(countername);
+
+            % add negative value to counter
+            ct.add(-1);
+            pause(70);
+
+            % fetch results
+            results = readJsonResults(testCase);
+            results = results{1};
+            dp = results.resourceMetrics.scopeMetrics.metrics.sum.dataPoints;
+
+            % verify that the counter value is still 0
+            verifyEqual(testCase, dp.asDouble, 0);
+
+        end
+
+
+        function testUpDownCounterBasic(testCase)
+            % test names and added value in UpDownCounter
+
+            metername = "foo";
+            countername = "bar";
+
+            p = opentelemetry.sdk.metrics.MeterProvider();
+            mt = p.getMeter(metername);
+            ct = mt.createUpDownCounter(countername);
+
+            % verify MATLAB object properties
+            verifyEqual(testCase, mt.Name, metername);
+            verifyEqual(testCase, mt.Version, "");
+            verifyEqual(testCase, mt.Schema, "");
+            verifyEqual(testCase, ct.Name, countername);
+
+            % create testing value 
+            val = 10;
```

Comment:
For UpDownCounter, it makes more sense to test negative values rather than positive values.